### ISSUE TITLE
format / toString inconsistency

### ISSUE THREAD

**Issue by oscarfv** [noojee/money.dart#22]
```
      print('money: ${m.toString()} ${m.format("0.00")}');
```

output:

```
money: 23,55 € 0.23
```

**Comment by oscarfv** [comment]
My locale uses comma as decimal separator and period as thousands separator:

```
final euro = Currency.create('EUR', 2,
    symbol: '€', invertSeparators: true, pattern: '#.##0,00 S');
```

 It seems that that is confusing `format`, but I would expect that the pattern is locale-indepent:

```
      final m = Money.from(23.55, euro);
      print('money: ${m.toString()} ${m.format("0,00")}');
```
output:

```
money: 23,55 € 23,55
```

The problem I'm trying to solve is to transform the monetary value to a double (with period as decimal separator, without thousands separator). The documentation says to use `MoneyEncoder`, but it seems that it is not exported (dart complains about `Undefined class`).

**Comment by oscarfv** [comment]
For the time being, I worked around the `MoneyDecoder` problem adding

```
import "package:money2/src/money_data.dart";
import "package:money2/src/encoders.dart";
```

to my Dart/Flutter source file.

**Comment by bsutton** [comment]
I've exposed money_data and encoders in the forthcoming 2.0 release.

However I'm a little confused about the problem.

You show an output of: 
money: 23,55 € 23,55

Is this not correct? - comma for decimal place.

Can you show me what you expect the output to be?

**Comment by oscarfv** [comment]
The documentation of `format` says:

```
  * , (comma) a placeholder for the grouping separtor
  * . (period) a place holder fo rthe decimal separator
```

but that's not what actually happens: the grouping separator and decimal separator on `format`'s pattern depends on the `invertSeparators` parameter of `Currency.create`. (FWIW, my preference is to adapt `format` to the current documentation, not to adapt the documentation to what `format` currently does).

Then noticed that `format` is the wrong approach to do what I want, because `format` will always use the comma as the decimal separator (because my `Currency` sets `invertSeparators = true`) and I need a string representation with a period as decimal separator (and without grouping separators). The right thing for this is to use `MoneyEncoder`.

To recap: this bug is about `format` not doing what its documentation says.

**Comment by bsutton** [comment]
OK, I think I understand what you are saying.

As it stands, in the format pattern you must always use a '.' for the decimal separator however when formatting it actually uses the currencies defined decimal separator.

This was the original design intent.

So there are four ways we could possibly fix this.

1) update the document to note your issue.
2) update the format so if invertSeparator is enabled then the format expects ',' for the decimal and '.' for groups.
3) change the behaviour of format so that it ignores the 'invertSeparators' field on the currency 
4) use an alternate currency.

So I don't think 3 is viable as when parsing the format pattern we would some how have to guess that you want the '.' and ',' to ignore the inversion. A pattern like '#.#' would be ambiguous in this case. 
The only way we could do this would be to add a flag to the format method to explicitly set the invertSeparators flag overwriting the currencies setting.

No. 4 however may actually be the correct approach as I think you requirement is probably unusual.

I think the following should work.

```dart
final notEuro = Currency.create('EUR', 2, symbol: '€', pattern: '###0.00 S');
final euro = Currency.create('EUR', 2,
    symbol: '€', invertSeparators: true, pattern: '#.##0,00 S');

final m = Money.from(23.55, euro);
final m2 = Money.fromBigInt(m.minorUnits, notEuro);

 print('money: ${m.toString()} ${m2.format("0,00")}');

```

**Comment by oscarfv** [comment]
My requirement is fulfilled by `MoneyEncoder`, this issue is no longer related to my original needs, but to the more general problem that I found: the documentation of `format` not reflecting what the function does.

So my preferred solution is 3, but if that is not acceptable (because the risk of breaking existing user code) doing 1 *and* 2 are the right thing IMHO to avoid the pitfall that prompted this issue.

BTW, the problem is not only about me trying to misuse `format`, users working with a `Currency` that sets `inverSeparators=true` and writing a pattern as per the specification described in `format` would likely face the same problem.
